pkg/installation/install.go

```diff
@@ -143,12 +147,20 @@ func createOrUpdateLink(binDir string, binary string, plugin string) error {
 	return nil
 }
 
-func removeLink(binDir string, plugin string) error {
-	dst := filepath.Join(binDir, pluginNameToBin(plugin))
-	if err := os.Remove(dst); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("failed to remove the symlink in %q, err: %v", dst, err)
-	} else if err == nil {
-		glog.V(3).Infof("Removed symlink from %q", dst)
+// removeLink removes a symlink reference if exists.
+func removeLink(path string) error {
+	if fi, err := os.Lstat(path); err == nil {
+		if fi.Mode()&os.ModeSymlink == 0 {
+			return fmt.Errorf("file %q is not a symlink (mode=%s)", path, fi.Mode())
+		}
+		if err := os.Remove(path); err != nil {
+			return fmt.Errorf("failed to remove the symlink in %q, err: %v", path, err)
+		}
+		glog.V(3).Infof("Removed symlink from %q", path)
+	} else if !os.IsNotExist(err) {
```

Comment:
Can we please invert the case. !...Not..() is quite difficult to understand

Reply:
Agreed we can unfold and revert this if a bit.